VirtoCommerce.CatalogModule.Web/Controllers/Api/CatalogModuleCategoriesController.cs

```diff
@@ -170,5 +175,39 @@ public IHttpActionResult Delete([FromUri]string[] ids)
             _categoryService.Delete(ids);
             return StatusCode(HttpStatusCode.NoContent);
         }
+
+        /// <summary>
+        /// Bulk deletes the specified categories by id.
+        /// </summary>
+        /// <param name="searchCriteria"></param>
+        [HttpPost]
+        [Route("delete")]
+        [ResponseType(typeof(void))]
+        public async Task<IHttpActionResult> BulkDelete(CategorySearchCriteria searchCriteria)
+        {
+            bool hasSearchCriteria;
+            var listIds = new List<string>();
+            do
+            {
+                var searchResult = await _categorySearchService.SearchAsync(searchCriteria);
```

Comment:
Here we are using indexed search to search for the categories. I think we should not use it at least if we have ObjectIds set. In this case, we should skip search and try to delete by the given ids.

Reply:
Changed